src/nvim/os/env.c

```diff
@@ -427,6 +427,31 @@ char *vim_getenv(const char *name)
     return xstrdup(kos_env_path);
   }
 
+  // XDG Base Directory expansion.
+  if (strncmp(name, "XDG_", 4) == 0) {
```

Comment:
As it already expand $HOME and others, I thought it was appropriate, but I don't mind changing. I should have updated the doc, though.